### PR TITLE
ENH: Add dta 119 reading to StataReader

### LINKED ISSUE #28250: Add support for reading Stata .dta file format 119
Please add functionality to read format 119 `.dta` Stata files! See file format here: https://www.stata.com/help.cgi?dta#versions

Currently, even though Pandas can _write_ format 119 files (https://github.com/pandas-dev/pandas/blob/612d3b23da5b99f6c5642be574fb08713a45d7d1/pandas/io/stata.py#L2663), it seems unable to _read_ format 119 files (https://github.com/pandas-dev/pandas/blob/612d3b23da5b99f6c5642be574fb08713a45d7d1/pandas/io/stata.py#L49-L53).

### PR DESCRIPTION
**Pull request by bashtage**
Add requirements for reading 119 format files

- [X] closes #28250
- [X] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by bashtage** [comment]
Minor code changes needed to support reading Stata DTA format 119.  The hard part is testing since this format is only used when a file has more than 32K variables, and so the test file hits this size.  This test isn't that fast (~20s).

File: pandas/io/stata.py [hunk 1]
```diff
@@ -1261,7 +1265,7 @@ def _get_fmtlist(self):
 
     # Returns the label list
     def _get_lbllist(self):
-        if self.format_version >= 118:
+        if self.format_version in (118, 119):
```
Comment (WillAyd) [inline pandas/io/stata.py]: Was the `>=` before a mistake or is there a reason why this needed to change?

**Review by WillAyd — COMMENTED** [review]
Hmm OK. I assume the test length is just spent in reading that many rows?

File: pandas/io/stata.py (hunk 1 above)
Comment (bashtage) [reply pandas/io/stata.py]: I reverted these to all be >=.

**Comment by bashtage** [comment]
> Hmm OK. I assume the test length is just spent in reading that many rows?

Yes, 32,999 columns. Have marked it slow.

**Comment by jreback** [comment]
thanks @bashtage

**Pull request merged**